packages/shared/src/components/buttons/Button.tsx

```diff
@@ -21,34 +27,36 @@ export interface BaseButtonProps {
   absolute?: boolean;
 }
 
-export type ButtonProps<Tag extends keyof JSX.IntrinsicElements> =
-  BaseButtonProps &
-    JSX.IntrinsicElements[Tag] & {
-      innerRef?: LegacyRef<JSX.IntrinsicElements[Tag]>;
-    };
+export type AllowedTags = keyof Pick<JSX.IntrinsicElements, 'a' | 'button'>;
+export type AllowedElements = HTMLButtonElement | HTMLAnchorElement;
 
-// eslint-disable-next-line @typescript-eslint/no-redeclare
-export function Button<Tag extends keyof JSX.IntrinsicElements>({
-  loading,
-  pressed,
-  icon,
-  rightIcon,
-  buttonSize,
-  children,
-  tag: Tag = 'button',
-  innerRef,
-  className,
-  displayClass,
-  absolute,
-  ...props
-}: StyledButtonProps & ButtonProps<Tag>): ReactElement {
+export type ButtonProps<Tag extends AllowedTags> = BaseButtonProps &
+  HTMLAttributes<AllowedElements> &
+  JSX.IntrinsicElements[Tag];
+
+export default function ButtonComponent<TagName extends AllowedTags>(
```

Comment:
Why do we export both components? I don't think we need the non `forwardRef` version.
This is very confusing